utils/thapi_gen_source_callbacks.rb

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env ruby
+#
+# Take output of `babeltrace_thapi trace -c` and generate a C metababel
+# callbacks SOURCE type plugin that will generate the same event stream.
+#
+# For use in creating test cases for unit testing, with input trace files that
+# are easy to maintain.
+#
+# Does not yet handle complex struct args, so the output may not be identical when
+# roundtripping through babeltrace_thapi.
+
+require 'optparse'
+require 'yaml'
+require 'erb'
+require 'time'
+require 'set'
+
+SOURCE_TEMPLATE = <<~TEXT.freeze
+  /* Code generated by #{__FILE__} */
+
+  #include <babeltrace2/babeltrace.h>
+  #include <metababel/metababel.h>
+
+  static const void** EMPTY_ARRAY;
```

Comment:
maybe transform it as a var`{{empty array name}}` to avoid "magic number"

Reply:
`static const void** #{{array_name}};` 

And then define `array_name="EMPTY_ARRAY"`, so we don't harcode the name of EMPTY_ARRAY in 2 places.